chia/util/condition_tools.py

```diff
@@ -96,10 +96,6 @@ def created_outputs_for_conditions_dict(
 ) -> List[Coin]:
     output_coins = []
     for cvp in conditions_dict.get(ConditionOpcode.CREATE_COIN, []):
-        # TODO: check condition very carefully
-        # (ensure there are the correct number and type of parameters)
-        # maybe write a type-checking framework for conditions
-        # and don't just fail with asserts
```

Comment:
I removed this TODO because this PR implements this suggestion

Reply:
one noteworthy difference is that this comment suggests checking the correct number of condition arguments, but that ship has sailed. We allow garbage at the end of the argument list and it would be a consensus change to reject such programs

Reply:
It's better this way anyway, as now we could now soft-fork in, for example, a four-argument `AGG_SIG` if we figure out some way to use the four argument version to be more restrictive than the three-argument version.